firmware/arduino/chessboard/movementFunctions.ino

```diff
@@ -1,10 +1,58 @@
 bool moveDirect(int startCol, int startRow, int endCol, int endRow)
 {
-  // TODO:
-  // Must account for diagonal and straight movement cases
-  // Must also return true if the current position is the target position 
-  // (Since no change is needed)
-  return true;
+  uint8_t statusCodeResult;
+  uint8_t curCol, curRow;
+  int8_t deltaX, deltaY;
+  bool isSuccessful;
+
+  deltaX = endCol - startCol;
+  deltaY = endRow - startRow;
+
+  // If target point is equal to the start point
+  if (deltaX == 0  &&  deltaY == 0)
+    return true;
+
+  // Checks if the EM is aligned properly
+  if ((currPositionX % stepsPerUnitSpace) || (currPositionY % stepsPerUnitSpace))
+    return false;
+
+  curCol = currPositionX / (stepsPerUnitSpace * 8);
+  curRow = currPositionY / (stepsPerUnitSpace * 8);
+
+  // Make initial move to first position. Move diagonally since it's faster.
+  if (!moveDirect(curCol, curRow, startCol, startRow))
+    return false;
```

Comment:
You need a check here that does something like
```c
if ((currCol != startCol || currRow != startRow) && !moveDirect(currCol, currRow, startCol, startRow)) {
  return false;
}
```
otherwise this will recurse infinitely. The movement doesn't happen until after you call this function, so the position will never be updated/movement will never happen without this check.

Reply:
This may be an easier way to do it, but if you trace the recursion 3 times it should work correctly and not recurse infinitely.